### ISSUE TITLE
Data race detected in TestContinuousChangesSubscription

### ISSUE THREAD

**Issue by bbrks** [couchbase/sync_gateway#3726]
Seen in http://uberjenkins.sc.couchbase.com:8080/job/_sync-gateway_github_pipeline/job/sync_gateway/job/PR-3691/4

```
...

2018-08-31T09:00:18.504-07:00 [INF] SyncMsg: [689949de] #1102: Type:rev Id:foo-1101 Rev:1-abc Deleted:false  User:GUEST
2018-08-31T09:00:18.508-07:00 [INF] SyncMsg: [689949de] #1103: Type:rev Id:foo-1102 Rev:1-abc Deleted:false  User:GUEST
2018-08-31T09:00:18.511-07:00 [INF] Sync: [689949de] Sent 1 changes to client, from seq 1101.  User:GUEST
2018-08-31T09:00:18.513-07:00 [INF] SyncMsg: [689949de] #1104: Type:rev Id:foo-1103 Rev:1-abc Deleted:false  User:GUEST
==================
WARNING: DATA RACE
Write at 0x00c423241030 by goroutine 204:
  github.com/couchbase/go-blip.(*Message).responseComplete()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/go-blip/message.go:296 +0x117
  github.com/couchbase/go-blip.(*Sender).send.func1.1()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/go-blip/sender.go:82 +0x4a
  github.com/couchbase/go-blip.(*Message).asyncRead.func1()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/go-blip/message.go:345 +0xba

Previous read at 0x00c423241030 by goroutine 183:
  github.com/couchbase/go-blip.(*Message).Response()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/go-blip/message.go:203 +0x6b
  github.com/couchbase/sync_gateway/rest.(*blipHandler).sendBatchOfChanges()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/blip_sync.go:396 +0x3b2
  github.com/couchbase/sync_gateway/rest.(*blipHandler).sendChanges.func2()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/blip_sync.go:352 +0xc4
  github.com/couchbase/sync_gateway/rest.(*blipHandler).sendChanges.func3()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/blip_sync.go:373 +0x85d
  github.com/couchbase/sync_gateway/rest.generateChanges()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/changes_api.go:552 +0x846
  github.com/couchbase/sync_gateway/rest.generateBlipSyncChanges()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/changes_api.go:434 +0xdb
  github.com/couchbase/sync_gateway/rest.(*blipHandler).sendChanges()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/blip_sync.go:380 +0x83a
  github.com/couchbase/sync_gateway/rest.(*blipHandler).handleSubChanges.func1()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/blip_sync.go:314 +0x1ac

Goroutine 204 (running) created at:
  github.com/couchbase/go-blip.(*Message).asyncRead()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/go-blip/message.go:346 +0x1a7
  github.com/couchbase/go-blip.(*Sender).send.func1()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/go-blip/sender.go:83 +0x3b6
  github.com/couchbase/go-blip.(*messageQueue).pushWithCallback()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/go-blip/messagequeue.go:100 +0x3ca
  github.com/couchbase/go-blip.(*Sender).send()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/go-blip/sender.go:88 +0x15e
  github.com/couchbase/go-blip.(*Sender).Send()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/go-blip/sender.go:61 +0x7f
  github.com/couchbase/sync_gateway/rest.(*blipHandler).sendBatchOfChanges()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/blip_sync.go:395 +0x3a1
  github.com/couchbase/sync_gateway/rest.(*blipHandler).sendChanges.func2()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/blip_sync.go:352 +0xc4
  github.com/couchbase/sync_gateway/rest.(*blipHandler).sendChanges.func3()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/blip_sync.go:373 +0x85d
  github.com/couchbase/sync_gateway/rest.generateChanges()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/changes_api.go:552 +0x846
  github.com/couchbase/sync_gateway/rest.generateBlipSyncChanges()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/changes_api.go:434 +0xdb
  github.com/couchbase/sync_gateway/rest.(*blipHandler).sendChanges()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/blip_sync.go:380 +0x83a
  github.com/couchbase/sync_gateway/rest.(*blipHandler).handleSubChanges.func1()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/blip_sync.go:314 +0x1ac

Goroutine 183 (running) created at:
  github.com/couchbase/sync_gateway/rest.(*blipHandler).handleSubChanges()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/blip_sync.go:316 +0x7dc
  github.com/couchbase/sync_gateway/rest.userBlipHandler.func1()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/blip_sync.go:74 +0xbe
  encoding/json.(*decodeState).value()
      /root/.gvm/gos/go1.8.5/src/encoding/json/decode.go:402 +0x149
  encoding/json.(*decodeState).unmarshal()
      /root/.gvm/gos/go1.8.5/src/encoding/json/decode.go:184 +0x2e8
  encoding/json.Unmarshal()
      /root/.gvm/gos/go1.8.5/src/encoding/json/decode.go:104 +0x20a
  github.com/couchbase/sync_gateway/auth.(*Authenticator).getPrincipal.func1()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/auth/auth.go:115 +0x161
  github.com/couchbaselabs/walrus.(*WalrusBucket).Update.func1()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbaselabs/walrus/crud.go:493 +0x78
  github.com/couchbaselabs/walrus.(*WalrusBucket).WriteUpdate()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbaselabs/walrus/crud.go:479 +0x16b
  github.com/couchbaselabs/walrus.(*WalrusBucket).Update()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbaselabs/walrus/crud.go:496 +0xc7
  github.com/couchbase/sync_gateway/base.(*LeakyBucket).Update()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/base/leaky_bucket.go:93 +0x7e
  github.com/couchbase/sync_gateway/auth.(*Authenticator).getPrincipal()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/auth/auth.go:148 +0x1e6
  github.com/couchbase/sync_gateway/auth.(*Authenticator).GetUser()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/auth/auth.go:82 +0xa1
  github.com/couchbase/sync_gateway/db.(*Database).ReloadUser()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/db/database.go:576 +0x20f
  github.com/couchbase/sync_gateway/rest.userBlipHandler.func1()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/blip_sync.go:69 +0x65
  github.com/couchbase/sync_gateway/rest.(*blipSyncContext).register.func1()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/sync_gateway/rest/blip_sync.go:155 +0x247
  github.com/couchbase/go-blip.(*Context).dispatchRequest()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/go-blip/context.go:180 +0x1df
  github.com/couchbase/go-blip.(*receiver).getPendingRequest.func1()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/go-blip/receiver.go:253 +0x8c
  github.com/couchbase/go-blip.(*Message).asyncRead.func1()
      /var/jenkins/workspace/peline_sync_gateway_PR-3691-KZUWCMNVEWZ4JIC5R4QA2OMEOAG46OGY4WVOUVXNEE7LDQL4OLQA/godeps/src/github.com/couchbase/go-blip/message.go:345 +0xba
==================
2018-08-31T09:00:18.517-07:00 [INF] SyncMsg: [689949de] #1105: Type:rev Id:foo-1104 Rev:1-abc Deleted:false  User:GUEST
2018-08-31T09:00:18.519-07:00 [INF] Sync: [689949de] Sent 1 changes to client, from seq 1102.  User:GUEST
2018-08-31T09:00:18.521-07:00 [INF] SyncMsg: [689949de] #1106: Type:rev Id:foo-1105 Rev:1-abc Deleted:false  User:GUEST
2018-08-31T09:00:18.521-07:00 [INF] Sync: [689949de] Sent 1 changes to client, from seq 1103.  User:GUEST
2018-08-31T09:00:18.523-07:00 [INF] Sync: [689949de] Sent 1 changes to client, from seq 1104.  User:GUEST
2018-08-31T09:00:18.526-07:00 [INF] SyncMsg: [689949de] #1107: Type:rev Id:foo-1106 Rev:1-abc Deleted:false  User:GUEST

...

--- FAIL: TestContinuousChangesSubscription (4.95s)
	testing.go:610: race detected during execution of test
```

**Comment by bbrks** [comment]
Go 1.10.x's race detector seems to find this race a lot more than 1.8.x

Seen in `TestMissingNoRev`

- http://uberjenkins.sc.couchbase.com:8080/job/_sync-gateway_github_pipeline/job/sync_gateway/job/PR-3770/5/
- http://uberjenkins.sc.couchbase.com:8080/job/_sync-gateway_github_pipeline/job/sync_gateway/job/PR-3769/2/
- http://uberjenkins.sc.couchbase.com/view/Build/job/_sync-gateway_github_pipeline/job/sync_gateway/job/release%252F2.1.1/2/

Probably worth considering during the Iridum timeframe to avoid dev headaches

**Comment by adamcfraser** [comment]
Has been fixed w/ #3883
